File: ios/Classes/lib/notifications/models/NotificationImageModel.swift
//
//  NotificationImageModel.swift
//  awesome_notifications
//
//  Created by Healofz on 12/07/22.
//

import Foundation

public class NotificationImageModel : AbstractModel {
    var url: String?
    var description: String?
    
    public func fromMap(arguments: [String : Any?]?) -> AbstractModel? {
        if(arguments == nil){ return self }
       
        self.url = MapUtils<String>.getValueOrDefault(reference: Definitions.NOTIFICATION_IMAGE_URL, arguments: arguments)
        self.description = MapUtils<String>.getValueOrDefault(reference: Definitions.NOTIFICATION_IMAGE_DESCRIPTION, arguments: arguments)

        return self
    }
    
    public func toMap() -> [String : Any?] {
        var mapData:[String: Any?] = [:]
        
        if(url != nil) {mapData[Definitions.NOTIFICATION_IMAGE_URL] = self.url}
        if(description != nil) {mapData[Definitions.NOTIFICATION_IMAGE_DESCRIPTION] = self.description}
        
        return mapData
    }
    
    public func validate() throws {
    }
}

File: android/src/main/java/me/carda/awesome_notifications/notifications/models/NotificationImageModel.java
package me.carda.awesome_notifications.notifications.models;

import android.content.Context;

import java.util.HashMap;
import java.util.Map;

import me.carda.awesome_notifications.Definitions;
import me.carda.awesome_notifications.notifications.exceptions.AwesomeNotificationException;

public class NotificationImageModel extends AbstractModel{

    public String url;
    public String description;

    public NotificationImageModel(){}

    @Override
    public NotificationImageModel fromMap(Map<String, Object> arguments) {
        url = getValueOrDefault(arguments, Definitions.NOTIFICATION_IMAGE_URL, String.class);
        description = getValueOrDefault(arguments, Definitions.NOTIFICATION_IMAGE_DESCRIPTION, String.class);
        return this;
    }

    @Override
    public Map<String, Object> toMap() {
        Map<String, Object> returnedObject = new HashMap<>();
        returnedObject.put(Definitions.NOTIFICATION_IMAGE_URL, url);
        returnedObject.put(Definitions.NOTIFICATION_IMAGE_DESCRIPTION, description);
        return returnedObject;
    }

    @Override
    public String toJson() {
        return templateToJson();
    }

    @Override
    public NotificationImageModel fromJson(String json) {
        return (NotificationImageModel) super.templateFromJson(json);
    }

    @Override
    public void validate(Context context) throws AwesomeNotificationException {

    }
}
